content/tutorials/build-apps/build-a-desktop-wallet-in-javascript.md

```diff
@@ -0,0 +1,1460 @@
+
+4. Now, add the following code to `preload.js`:
+
+```javascript
+const { contextBridge, ipcRenderer } = require('electron');
+
+contextBridge.exposeInMainWorld('electronAPI', {
+  onUpdateLedgerData: (callback) => {
+    ipcRenderer.on('update-ledger-data', callback)
+  },
+
+  // Step 3 code additions - start
+  onEnterAccountAddress: (address) => {
+    ipcRenderer.send('address-entered', address)
+  },
+  onUpdateAccountData: (callback) => {
+    ipcRenderer.on('update-account-data', callback)
+  }
+  //Step 3 code additions - end
+})
+```
+
+Here we can observe a notable difference to the previous step. Until now we just used `ipcRenderer` to pick up on events from the main logic, now we are using it bidirectional to send events from the frontend to the main logic:
+
+```javascript
+onEnterAccountAddress: (address) => {
+    ipcRenderer.send('address-entered', address)
+}
+```
+
+5. Then, modify the `view/template.html` by replacing the `<body>` with this markup: 
+
+```html
+<body>
+
+    <h3>Build a XRPL Wallet - Part 3/8</h3>
+
+    <fieldset>
+        <legend>Account info</legend>
+        Classic Address: <strong id="account-address-classic"></strong><br/>
+        X-Address: <strong id="account-address-x"></strong><br/>
+        XRP Balance: <strong id="account-balance"></strong><br/>
+        XRP Reserved: <strong id="account-reserve"></strong><br/>
+    </fieldset>
+
+    <fieldset>
+        <legend>Latest validated ledger stats</legend>
+        Ledger Index: <strong id="ledger-index"></strong><br/>
+        Ledger Hash: <strong id="ledger-hash"></strong><br/>
+        Close Time: <strong id="ledger-close-time"></strong><br/>
+    </fieldset>
+
+    <dialog id="account-address-dialog">
+        <form method="dialog">
+            <div>
+                <label for="address-input">Enter account address:</label>
+                <input type="text" id="address-input" name="address-input" />
+            </div>
+            <div>
+                <button type="submit">Confirm</button>
+            </div>
+        </form>
+    </dialog>
+
+</body>
+```
+
+6. To incorporate the refactored markup, handle the HTML dialog element and well as the new account data section replace the contents of `view/renderer.js` with the following code:
+
+{{ include_code("_code-samples/build-a-wallet/desktop-js/3-account/view/renderer.js", language="js") }}
+
+Now you need an XRPL account address to monitor. If you already have one or know where to find an example, you can now run the application by executing:
+
+```console
+npm run start
+```
+If you are new to the XRPL and need an account address, [you can get accounts on the testnet](https://learn.xrpl.org/course/code-with-the-xrpl/lesson/create-accounts-and-send-xrp/). There you can also use the sandbox to issue XRP transactions, which then should show up in our app.
```

Comment:
I'd recommend linking to the faucet here instead of a separate tutorial. They mostly just need an account, and the faucet page let's them do that. Otherwise I may get confused and try to start the other tutorial mid-way here.
```suggestion
If you are new to the XRPL and need an account address, [you can get accounts on the testnet](https://xrpl.org/xrp-testnet-faucet.html). There you can also use the sandbox to issue XRP transactions, which then should show up in our app.
```